### DISCUSSION TITLE
Can't build .NET 6 from source

### DISCUSSION THREAD

**Discussion by federico-r-figueredo** [dotnet/source-build#3110] [General]
Greetings, I'm following all the steps to build .NET 6 from source in my Ubuntu 22.04.1 LTS and they succeed until I get to the step where I've to run this command...

![Screenshot from 2022-11-07 01-22-19](https://user-images.githubusercontent.com/49570839/200226075-c7a7cbd0-f265-4bd5-947e-cf276685fea5.png)

...and I get the following error...

![Screenshot from 2022-11-07 01-19-54](https://user-images.githubusercontent.com/49570839/200226034-6a0ca207-62bb-4cc6-8f61-ee38534e2ebe.png)

**Comment by marcpopMSFT** [comment]
Are you building the sdk repo or building .NET source build?
@dotnet/source-build-internal

**Reply by federico-r-figueredo** [reply]
I've built following the instructions from https://github.com/dotnet/installer after being redirected there from the instructions in https://github.com/dotnet/source-build

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by crummel** [comment]
This looks like building from the source-build tarball.  It looks like you might have a corrupt set of package archives - could you delete `packages/archive/*.tar.gz` and the `.dotnet` directory at the root and try `./prep.sh` and `./build.sh` again?  If that doesn't work, could I get the contents of `packages/archive/archiveArtifacts.txt`?

**Reply by federico-r-figueredo** [reply]
I've built following the instructions from https://github.com/dotnet/installer after being redirected there from the instructions in https://github.com/dotnet/source-build. Thx, I'll try that and fetch that .txt if it fails.

**Reply by federico-r-figueredo** [reply]
I did exactly that but got this error...

![image](https://user-images.githubusercontent.com/49570839/202887582-f4758367-f41b-439e-8784-320e31c7060c.png)

I assume it could be a specific problem on my machine closing the data stream.

Next, if I try to ./prep.sh again, it detects the .tar.gz already exists (but probably corrupted due to incomplete download)

![image](https://user-images.githubusercontent.com/49570839/202887673-6faa53e4-cca4-46e7-8e9e-d8ae54ccca67.png)

I'd like to contribute to the project but I'm having a hard time building from source.

**Reply by MichaelSimons** [reply]
@federico-r-figueredo - Since the download failed, you will have to manually delete the partial tarball at ./packages/archive/Private.SourceBuilt.Artifacts.*.tar.gz  I will log an issue for this bad UX.

**Reply by federico-r-figueredo** [reply]
@MichaelSimons  All right, so I manually downloaded successfully the tarball and placed it in its intended location...

![image](https://user-images.githubusercontent.com/49570839/205459879-81ca26dc-64a8-4544-b78a-636cb08bc509.png)

Now, with the "/prep.sh" step completed successfully (I've both the NET SDK & the tarball in their respective locations), I try to run the command of the next step "./build.sh" but I get this error...

![image](https://user-images.githubusercontent.com/49570839/205459956-beff4b21-8ecf-4002-bc5a-c2a9b337eecb.png)

**Reply by federico-r-figueredo** [reply]
@MichaelSimons  Could be the case, that the problem lies in the fact that the tarball seems to be 3.9 GB but my browser only downloaded 946 MB and marked it as download complete?

![image](https://user-images.githubusercontent.com/49570839/205460133-2cc13523-f66e-4a2b-b240-b29aeb7abc68.png)

**Reply by federico-r-figueredo** [reply]
I tried to download again and got a 1.1 GB only tarball. Is there any other URL from which we can download it? Seems **https://dotnetcli.azureedge.net/source-built-artifacts/assets/Private.SourceBuilt.Artifacts.6.0.107.tar.gz** keeps failing again and again.

**Reply by MichaelSimons** [reply]
This is the only location to download the tarball from.  FYI we have addressed the size of this tarball starting in 7.0.  It is now 1.2 GB.  We have plans to reduce this even further in .net 8.0
